### PR TITLE
Websockets fix

### PR DESCRIPTION
**Pull request by AntoineDao**
Small fix to what was causing the server to break when broadcasting messages to a room.

On the topic of Websockets. Is there anywhere I can find:

1. What message payload is expected to look like? There is a chance my fix is pointless is I just missed a crucial variable
2. What the routine of connecting/disconnecting is? I notice that you can `join` and `leave` rooms but am not sure what those are for.
3. Is it normal that I get disconnected after about a minute? it seems `ws.on('close')` event happens for some reason and I get bumped off the server.

### CODE REVIEW

**Comment by AntoineDao** [comment]
2nd time does the trick!

**Comment by AntoineDao** [comment]
No urgency to review don't worry. Just figured I'd ping some victim to have a quick look when they have a minute 😄

**Comment by didimitrie** [comment]
> What message payload is expected to look like? There is a chance my fix is pointless is I just missed a crucial variable

I found some test objects:

Join project room:

```js
{"eventName":"join","resourceId":"5bf5ac67cff82a0a0860d7e0", "resourceType":"project"}
```

Leave project room:

```js
{"eventName":"leave","resourceId":"5bf5ac67cff82a0a0860d7e0", "resourceType":"project"}
```

```js
{"eventName":"leave","roomName":"project-5bf5ac67cff82a0a0860d7e0"}
```

Broadcast: 
```js
{"eventName":"broadcast","streamId":"project-5bf5ac67cff82a0a0860d7e0"}
```

A custom message needs to be passed in the `args` field, ie:

```cs
var eventData = new
      {
        eventName = "broadcast",
        senderId = ClientId,
        resourceType = resourceType,
        resourceId = resourceId,
        args = args <--- That's a dynamic object, can be anything that can be "jsonified"
      };
```

> What the routine of connecting/disconnecting is? I notice that you can join and leave rooms but am not sure what those are for.

Each resource is a "chat room" that a client can join and listen to any events being broadcast in there. See it like a cheap version of firebase, without the scalability.

> Is it normal that I get disconnected after about a minute? it seems ws.on('close') event happens for some reason and I get bumped off the server.

Most probably you don't send a ping back. I couldn't get ws heartbeats to work reliably in .net, so there's a custom ping event being sent out. Here's the implementation in SpeckleCore:

https://github.com/speckleworks/SpeckleCore/blob/master/ApiClient/SpeckleApiClientExtension.cs#L266

Ie, just send an alive if you get a ping...

---

So now my question: what was the  borkage? Me can't grok from the [diff](https://github.com/speckleworks/SpeckleServer/pull/167/files), but my nose is blocked and hence there's little oxygen getting to my brain these days...

File: app/ws/RadioTower.js [hunk 1]
```diff
@@ -85,13 +85,14 @@ module.exports = {
...
-      } else if ( !( !message.resourceId || !message.resourceType ) )
+      } else if ( !( !message.resourceId || !message.resourceType ) ) {
         roomName = `${message.resourceType}-${message.resourceId}`
+      }
 
-      if ( roomName === '' )
+      if ( roomName === '' ) {
         return winston.debug( `Failed to deliver broadcast from ${senderClientId} (no room name).` )
-
-      winston.debug( `📣 broadcast in ${roomName} from ${senderClientId}: ${message.args.eventType}` )
```
Comment (AntoineDao) [inline app/ws/RadioTower.js]: This is where I saw a bug. After reading your comments I'm still not sure whether args should have an event type as it's mean to be an `anything that can be jsonified` object. Regardless, seems a shame that a debug log would kill a process from running 😄

Comment (didimitrie) [reply app/ws/RadioTower.js]: Awwww yes 🙈

**Pull request merged**